sources/AppBundle/Openfeedback/OpenfeedbackJsonGenerator.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace AppBundle\Openfeedback;
+
+use AppBundle\CFP\PhotoStorage;
+use AppBundle\Event\Model\Event;
+use AppBundle\Event\Model\Planning;
+use AppBundle\Event\Model\Repository\TalkRepository;
+use AppBundle\Event\Model\Room;
+use AppBundle\Event\Model\Speaker;
+use AppBundle\Event\Model\Talk;
+
+class OpenfeedbackJsonGenerator
+{
+    /**
+     * @var TalkRepository
+     */
+    private $talkRepository;
+
+    /**
+     * @var PhotoStorage
+     */
+    private $photoStorage;
+
+    /**
+     * @param TalkRepository $talkRepository
+     * @param PhotoStorage $photoStorage
+     */
+    public function __construct(TalkRepository $talkRepository, PhotoStorage $photoStorage)
+    {
+        $this->talkRepository = $talkRepository;
+        $this->photoStorage = $photoStorage;
+    }
+
+    /**
+     * @param Event $event
+     *
+     * @return array
+     */
+    public function generate(Event $event)
+    {
+        $talks = $this->talkRepository->getByEventWithSpeakers($event);
+        $data = [];
+        foreach ($talks as $talkWithData) {
+            /**
+             * @var Talk
+             */
+            $talk = $talkWithData['talk'];
+
+            /**
+             * @var Speaker[]
+             */
+            $speakers = $talkWithData['.aggregation']['speaker'];
+
+            /**
+             * @var Planning
+             */
+            $planning = $talkWithData['planning'];
+
+            /**
+             * @var Room
+             */
+            $room = $talkWithData['room'];
+
+            $speakersFormatted = [];
+            $speakersId = [];
+            foreach ($speakers as $speaker) {
+                $speakersId[] = "{$speaker->getId()}";
+                $speakersFormatted[] = [
+                    'name' => $speaker->getLabel(),
+                    'photoUrl' => 'https://afup.org' . $this->photoStorage->getUrl($speaker),
+                    'social' => [
+                        [
+                            'Présentation',
+                            'link' => sprintf('https://event.afup.org/%s/speakers/#%d', $event->getPath(), $speaker->getId()),
+                        ]
+                    ],
+                    'id' => "{$speaker->getId()}",
+                ];
+            }
+
+            $talkFormatted = [
+                'speakers' => $speakersId,
+                'tags' => [$talk->getTypeLabel()],
+                'title' => $talk->getTitle(),
+                'id' => "{$talk->getId()}",
+                'startTime' => $this->getOpenfeedbackFormat($planning->getStart()),
+                'endTime' => $this->getOpenfeedbackFormat($planning->getEnd()),
+                'trackTitle' => ($room ? $room->getName() : ''),
+            ];
+
+            $data['sessions'][$talk->getId()] = $talkFormatted;
+
+            if ($this->isNotSpeakerDuplicatation($data, $speakersFormatted)) {
```

Comment:
pourquoi y a t-il besoin de ce test, le fait que la clef soit "$person['id']" ne suffit pas à dédoublonner ?

Reply:
Non sinon à chaque passage dans la boucle on le push dans le tableau et si on a un speaker qui fait plusieurs conférences on a des doublons

Reply:
ce que je ne comprends pas c'est qu'il aura le même id pour un événement même si il est lié à plusieurs sessions.
(par contre effectivement le même speaker aura des ids différents entre les événements, mais ici on va toujours envoyer un seul événement)

Reply:
En faite on a un tableau ici speakers qui doit contenir la liste des speakers (de la même manière que l'on a un tableau qui contient au même niveau la liste des sessions). L'id du speaker est fixe et celui de l'évènement aussi. L'id speaker permet de faire référence entre l'objet session et speaker pour faire le lien. Ce test sert juste à ne pas avoir de doublon dans le tableau des speakers si on a plusieurs sessions animées par un même speakers. A mon sens c'est inévitable car on boucle avec un foreach sur la liste des talks de l'event.

Reply:
Done ✅